iOS10-HW-12-Rybkin-Ivan/Soursers/ViewControllers/ViewController.swift

```diff
@@ -0,0 +1,245 @@
+            startButton.setImage(UIImage(systemName: "play"), for: .normal)
+            if !isWorkTime {
+                startButton.tintColor = .systemGreen
+                timeLabel.textColor = .systemGreen
+            }
+        } else {
+            if isFirstPlay {
+                isStarted = true
+                stateLabel.text = "Work"
+                stateLabel.textColor = .systemRed
+                startButton.imageView?.tintColor = .systemRed
+                startButton.setImage(UIImage(systemName: "pause"), for: .normal)
+                timeLabel.textColor = .systemRed
+                animateCircularProgress(duration: TimeInterval(timeRemaining))
+                isFirstPlay = false
+            } else {
+                isStarted = true
+                startButton.setImage(UIImage(systemName: "pause"), for: .normal)
+                resumeAnimation()
+                if !isWorkTime {
+                    stateLabel.text = "Relax"
+                    stateLabel.textColor = .systemGreen
+                    startButton.imageView?.tintColor = .systemGreen
+                    timeLabel.textColor = .systemGreen
+                    circleLayer.strokeColor = UIColor.systemGreen.cgColor
+                } else {
+                    stateLabel.textColor = isResting ? .systemGreen : .systemRed
+                    startButton.imageView?.tintColor = isResting ? .systemGreen : .systemRed
+                }
+            }
+            timerSeting()
+        }
+    }
+
+    private func startTimer() {
+        stateLabel.text = "Work"
+        stateLabel.textColor = .systemRed
+        startButton.imageView?.tintColor = .systemRed
+        timeLabel.textColor = .systemRed
+        updateTimeLabel()
+        timerSeting()
+        resumeAnimation()
+        animateCircularProgress(duration: TimeInterval(timeRemaining))
+        isResting = false
+
+    }
+
+    private func startRest() {
+        stateLabel.text = "Relax"
+        stateLabel.textColor = .systemGreen
+        startButton.imageView?.tintColor = .systemGreen
+        timeLabel.textColor = .systemGreen
+        updateTimeLabel()
+        timerSeting()
+        resumeAnimation()
+        animateCircularProgress(duration: TimeInterval(timeRemaining))
+        isResting = true
+    }
+
+    private func handleTimerCompletion() {
+        pauseTimer()
+        if isResting {
+            isResting = false
+            timeRemaining = 15
+            startTimer()
+        } else {
+            isResting = true
+            timeRemaining = 5
+            startRest()
+        }
+
+    }
+
+    private func timerSeting() {
+        timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true, block: { [weak self] _ in
+            guard let self = self else { return }
+            self.timeRemaining -= 1
+            self.updateTimeLabel()
+            if self.timeRemaining == 0 {
+                self.handleTimerCompletion()
+            }
+        })
```

Comment:
молодец что ослабил ссылку
```suggestion
      timer = Timer.scheduledTimer(
            withTimeInterval: 1,
            repeats: true, block: { [weak self] _ in
            guard let self = self else { return }
            self.timeRemaining -= 1
            self.updateTimeLabel()
            if self.timeRemaining == 0 {
                self.handleTimerCompletion()
            }
        })
```